util/easing/src/easing.c

```diff
@@ -43,12 +43,12 @@ static inline float exponential_out(float step, float max_steps, float max_val)
 {
 	return (step == max_steps) ?
         max_val :
-        max_val - pow(max_val, 1.0 - (float)step/max_steps);
+        max_val - pow(max_val, 1.0f - (float)step/max_steps);
```

Comment:
Since you are moving everything to `float` you could as well get rid of those `(float)` casts. This one would be the same as `max_val - pow(max_val, 1.0f - step/max_steps)`. This happens across much of the refactoring.

Reply:
I'm not sure why the original implementation had that cast there, it was promoted to a `double` anyway. I presume that early on `step` and `max_step` were integer values and it was a remnant of refactoring. I can certainly clean those up.

Reply:
The indentations where inherited from a previous implementation I had while validating this mathematically.